src/lib/store.js

```diff
@@ -483,4 +526,49 @@ export default class Store {
 		this.customPurposeList = customPurposeList;
 		this.storeUpdate();
 	};
+
+	mergeVendorConsentsFromGlobalCookie = () => {
+		let selectedLocalIdsOfGlobalVendorInGlobalCookie = new Set();
+		let selectedLocalIdsOfGlobalVendorInLocalCookie = new Set();
+		const globalCookieMaxVendorId = this.persistedGlobalVendorConsentData.maxVendorId;
+
+		this.globalVendorIdsPresentOnList.forEach( (localId, globalId) => {
+			if (this.vendorConsentData.selectedVendorIds.has(localId) && globalId <= globalCookieMaxVendorId) {
+				selectedLocalIdsOfGlobalVendorInLocalCookie.add(localId);
+			}
+			if (this.globalVendorConsentData.selectedVendorIds.has(globalId)) {
+				selectedLocalIdsOfGlobalVendorInGlobalCookie.add(localId);
+				this.vendorConsentData.selectedVendorIds.add(localId);
+			}
+		});
+
+		selectedLocalIdsOfGlobalVendorInLocalCookie.forEach( localId => {
+			const isNewVendorOnList = this.persistedVendorConsentData && localId > this.persistedVendorConsentData.maxVendorId;
+
+			if (!isNewVendorOnList && !selectedLocalIdsOfGlobalVendorInGlobalCookie.has(localId)) {
+				this.vendorConsentData.selectedVendorIds.delete(localId);
+			}
+		});
+	};
+
+	mergeVendorConsentsToGlobalCookie = () => {
+		this.globalVendorIdsPresentOnList.forEach( (localId, globalId) => {
+			if (this.vendorConsentData.selectedVendorIds.has(localId)) {
+				this.globalVendorConsentData.selectedVendorIds.add(globalId);
+			} else {
+				this.globalVendorConsentData.selectedVendorIds.delete(globalId);
+			}
+		});
+	};
+
+	mergePurposeConsentsFromGlobalCookie = () => {
+		const selectedPurposeIds = Array.from(this.vendorConsentData.selectedPurposeIds);
+		this.vendorConsentData.selectedPurposeIds = new Set(selectedPurposeIds.filter( id => (
+			this.globalVendorConsentData.selectedPurposeIds.has(id) || config.legIntPurposeIds.includes(id)
+		)));
+	};
+
+	mergePurposeConsentsToGlobalCookie = () => {
+		this.globalVendorConsentData.selectedPurposeIds = new Set(...this.vendorConsentData.selectedPurposeIds);
```

Comment:
`this.vendorConsentData.selectedPurposeIds` is a Set, so we can't use spread operator. It should be
 `... = this.vendorConsentData.selectedPurposeIds`.

Reply:
In addition syntax of Set constructor is `new Set([iterable]);`